src/ntp/client.rs

```diff
@@ -40,9 +40,7 @@ pub enum NtpClientError {
 
 impl std::error::Error for NtpClientError {
     fn description(&self) -> &str {
-        match self {
-            _ => "Connection to server failed because address could not be resolved",
-        }
+        "Connection to server failed because address could not be resolved"
```

Comment:
Was this clippy? This original code looks a little strange, but I wonder if the intention was that the string returned would depend on the variant of `self`. We could do this instead:
```rust
match self {
    Self::NoIpV4AddrFound | Self::NoIpv6AddrFound | Self::InvalidUid => "XXX",
}
```
That way the compiler will force us to deal with variants explicitly.

Also, we might consider making the description actually reflect the variant :)

Reply:
Similarly below.